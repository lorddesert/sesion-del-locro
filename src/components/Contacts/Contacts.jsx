import React, { useState, useCallback, useEffect, useContext } from "react"
import Context from '../../context/GlobalContext'
import "./Contacts.scss"

// Components
import Contact from "../Contact/Contact"
import SecondaryButton from "../SecondaryButton/SecondaryButton"
import ChatRoom from "../ChatRoom/ChatRoom"

import altUserImg from "./resources/altuser.png"
import add from "./resources/add.svg"

const Contacts = props => {

  const globalContext = useContext(Context)
  const { app, auth } = globalContext

  const [contacts, setContacts] = useState([])

  // useEffect(() => {

  //   //*setContacts
  //   (async () => {
  //     try {
  //       let contacts = [];
  //       const usersRef = app.database().ref("users");
  //       const sender = auth.currentUser.uid;
  //       const myNickname = auth.currentUser.displayName;
  
  //       let snapshot = await usersRef.once("value");
  
  //       snapshot.forEach((user) => {
  //         if (user.val().nickname !== myNickname) {
  //           const { photo, online, nickname } = user.val();
  //           let chat = user.child(`contacts/${sender}/chat`).val();
  
  //           chat = chat ? Object.values(chat) : [];
  
  //           const newContact = {
  //             chat,
  //             photo,
  //             online,
  //             nickname,
  //             ref: user,
  //           };
  
  //           contacts.push(newContact);
  //         }
  //       });
  
  //       setContacts(contacts);
  //     } catch (error) {
  //       console.log(error);
  //     }})()

  // }, [])

  const [chatRooms, setChatRooms] = useState([])

  // useEffect(() => {

  //   //*setChatRooms
  //   (async () => {
  //     try {
  //       const chatRoomRef = app.database().ref("chatRooms");
  //       const snapshot = await chatRoomRef.once("value");
  //       let chatRooms = [];
  
  //       snapshot.forEach((chatRoom) => {
  //         const newChatRoom = {
  //           ...chatRoom.val(),
  //           chat: chatRoom.hasChild("chat")
  //             ? Object.values(chatRoom.child("chat").val())
  //             : [],
  //         };
  
  //         chatRooms.push(newChatRoom);
  //       });
  
  //       setChatRooms(chatRooms)
  //     } catch (error) {
  //       console.log(error);
  //     }
  //   })()

  // }, [])

  const modal = useCallback(() => {
    const modal = document.getElementById("modal")
    const modalForm = document.getElementById("modalForm")

    setTimeout(() => modalForm.classList.toggle("modalTransition"), 10)
    modal.classList.toggle("show-modal")
  })

  const addContact = useCallback(() => {
    alert("añadir modal para agregar contacto:\nnombre de usuario?\nnickname?")
  })

  const toggleRotate = useCallback(() => {
    let userCfg = document.getElementById("userCfg")
    let userCfgPanel = document.getElementById("userCfgPanel")

    userCfg.classList.toggle("rotate")
    const containsDisplay = userCfgPanel.classList.contains("display")

    if (!containsDisplay) {
      userCfgPanel.style.display = "grid"
      setTimeout(() => userCfgPanel.classList.toggle("display"), 100)
    } else {
      userCfgPanel.classList.toggle("display")
      setTimeout(() => (userCfgPanel.style.display = "none"), 100)
    }
  })

  const toggleUserConfig = useCallback(() => {
    this.toggleRotate()
    setTimeout(() => {
      const ref = document.getElementById("userConfig")
      ref.classList.toggle("toggleUserConfig")
    }, 150)
  })

  const disconnect = useCallback(() => {
    this.props.auth.signOut()
    window.location.reload()
  })

  const displayContacts = useCallback(() => {
    const contacts = document.getElementById("contacts-display")
    const contactsArrow = document.getElementById("contacts-arrow")

    contacts.style.display = "flex"
    contacts.style.alignItems = "start"
    contactsArrow.style.transition = "all ease 150ms"
    contactsArrow.style.transform = "rotate(0deg)"
  })

  const dummyFn = () => alert("asd")
  
    if(contacts.length){
    return (
      <div className="Contacts" id="contacts">
        <div className="Contacts-content">
          <div className="Profile">
            <div className="Profile-img">
              {/* {props.user.photo ? (
                <img
                  // src={props.user.photo || ''}
                  src={''}
                  // alt={`${props.user.nickname || ''}`}
                  alt={`${''}`}
                ></img>
              ) : (
                <img
                  className="alt-user-img"
                  // src={altUserImg || ''}
                  // alt={props.user.nickname || 'default'}
                  src={''}
                  alt={'default'}
                />
              )} */}
            </div>
            <div className="Profile-actions">
              <div className="Action" onClick={toggleRotate}>
                <svg
                  id="userCfg"
                  className="Action-img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path d="m499.953125 197.703125-39.351563-8.554687c-3.421874-10.476563-7.660156-20.695313-12.664062-30.539063l21.785156-33.886719c3.890625-6.054687 3.035156-14.003906-2.050781-19.089844l-61.304687-61.304687c-5.085938-5.085937-13.035157-5.941406-19.089844-2.050781l-33.886719 21.785156c-9.84375-5.003906-20.0625-9.242188-30.539063-12.664062l-8.554687-39.351563c-1.527344-7.03125-7.753906-12.046875-14.949219-12.046875h-86.695312c-7.195313 0-13.421875 5.015625-14.949219 12.046875l-8.554687 39.351563c-10.476563 3.421874-20.695313 7.660156-30.539063 12.664062l-33.886719-21.785156c-6.054687-3.890625-14.003906-3.035156-19.089844 2.050781l-61.304687 61.304687c-5.085937 5.085938-5.941406 13.035157-2.050781 19.089844l21.785156 33.886719c-5.003906 9.84375-9.242188 20.0625-12.664062 30.539063l-39.351563 8.554687c-7.03125 1.53125-12.046875 7.753906-12.046875 14.949219v86.695312c0 7.195313 5.015625 13.417969 12.046875 14.949219l39.351563 8.554687c3.421874 10.476563 7.660156 20.695313 12.664062 30.539063l-21.785156 33.886719c-3.890625 6.054687-3.035156 14.003906 2.050781 19.089844l61.304687 61.304687c5.085938 5.085937 13.035157 5.941406 19.089844 2.050781l33.886719-21.785156c9.84375 5.003906 20.0625 9.242188 30.539063 12.664062l8.554687 39.351563c1.527344 7.03125 7.753906 12.046875 14.949219 12.046875h86.695312c7.195313 0 13.421875-5.015625 14.949219-12.046875l8.554687-39.351563c10.476563-3.421874 20.695313-7.660156 30.539063-12.664062l33.886719 21.785156c6.054687 3.890625 14.003906 3.039063 19.089844-2.050781l61.304687-61.304687c5.085937-5.085938 5.941406-13.035157 2.050781-19.089844l-21.785156-33.886719c5.003906-9.84375 9.242188-20.0625 12.664062-30.539063l39.351563-8.554687c7.03125-1.53125 12.046875-7.753906 12.046875-14.949219v-86.695312c0-7.195313-5.015625-13.417969-12.046875-14.949219zm-152.160156 58.296875c0 50.613281-41.179688 91.792969-91.792969 91.792969s-91.792969-41.179688-91.792969-91.792969 41.179688-91.792969 91.792969-91.792969 91.792969 41.179688 91.792969 91.792969zm0 0" />
                </svg>
              </div>
            </div>
            <div className="userCfgPanel" id="userCfgPanel">
              <div className="panelOption config">
                <SecondaryButton
                  value="Ajustes de usuario"
                  action={toggleUserConfig}
                />
              </div>
              <div className="panelOption logOut">
                <SecondaryButton
                  value="Cerrar Sesión"
                  action={disconnect}
                />
              </div>
            </div>
          </div>
          {/* onClick will be changed with "enterChatRoom"  */}

          {/* // !FROM HERE */}

          {contacts && contacts.map((contact, i) => (
            <Contact
              key={`contact-${i}`}
              contact={contact || {}}
              // setChat={props.setChat}
              setChat={dummyFn}
              number={i}
            />
          ))}
          <div className="chat-room">
            <span>Salas de chat</span>
            <div
              className="image-wrapper"
              onClick={() => props.toggleModal && props.toggleModal(true)}
            >
              <img src={add} alt="crear sala de chat"></img>
            </div>
          </div>
          {chatRooms.length &&
            chatRooms.map((chatRoom, i) => (
              <ChatRoom
                key={`chatRoom-${i}`}
                chatRoom={chatRoom}
                setChatRoom={props.setChatRoom}
                number={i}
              />
            ))}
        </div>
      </div>
    )}

    else {
        return <div className="Contacts" id="contacts">
        <div className="Contacts-content">
          <div className="Profile">
            <div className="Profile-img">
              {/* {props.user.photo ? (
                <img
                  // src={props.user.photo || ''}
                  src={''}
                  // alt={`${props.user.nickname || ''}`}
                  alt={`${''}`}
                ></img>
              ) : (
                <img
                  className="alt-user-img"
                  // src={altUserImg || ''}
                  // alt={props.user.nickname || 'default'}
                  src={''}
                  alt={'default'}
                />
              )} */}
            </div>
            <div className="Profile-actions">
              <div className="Action" onClick={toggleRotate}>
                <svg
                  id="userCfg"
                  className="Action-img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path d="m499.953125 197.703125-39.351563-8.554687c-3.421874-10.476563-7.660156-20.695313-12.664062-30.539063l21.785156-33.886719c3.890625-6.054687 3.035156-14.003906-2.050781-19.089844l-61.304687-61.304687c-5.085938-5.085937-13.035157-5.941406-19.089844-2.050781l-33.886719 21.785156c-9.84375-5.003906-20.0625-9.242188-30.539063-12.664062l-8.554687-39.351563c-1.527344-7.03125-7.753906-12.046875-14.949219-12.046875h-86.695312c-7.195313 0-13.421875 5.015625-14.949219 12.046875l-8.554687 39.351563c-10.476563 3.421874-20.695313 7.660156-30.539063 12.664062l-33.886719-21.785156c-6.054687-3.890625-14.003906-3.035156-19.089844 2.050781l-61.304687 61.304687c-5.085937 5.085938-5.941406 13.035157-2.050781 19.089844l21.785156 33.886719c-5.003906 9.84375-9.242188 20.0625-12.664062 30.539063l-39.351563 8.554687c-7.03125 1.53125-12.046875 7.753906-12.046875 14.949219v86.695312c0 7.195313 5.015625 13.417969 12.046875 14.949219l39.351563 8.554687c3.421874 10.476563 7.660156 20.695313 12.664062 30.539063l-21.785156 33.886719c-3.890625 6.054687-3.035156 14.003906 2.050781 19.089844l61.304687 61.304687c5.085938 5.085937 13.035157 5.941406 19.089844 2.050781l33.886719-21.785156c9.84375 5.003906 20.0625 9.242188 30.539063 12.664062l8.554687 39.351563c1.527344 7.03125 7.753906 12.046875 14.949219 12.046875h86.695312c7.195313 0 13.421875-5.015625 14.949219-12.046875l8.554687-39.351563c10.476563-3.421874 20.695313-7.660156 30.539063-12.664062l33.886719 21.785156c6.054687 3.890625 14.003906 3.039063 19.089844-2.050781l61.304687-61.304687c5.085937-5.085938 5.941406-13.035157 2.050781-19.089844l-21.785156-33.886719c5.003906-9.84375 9.242188-20.0625 12.664062-30.539063l39.351563-8.554687c7.03125-1.53125 12.046875-7.753906 12.046875-14.949219v-86.695312c0-7.195313-5.015625-13.417969-12.046875-14.949219zm-152.160156 58.296875c0 50.613281-41.179688 91.792969-91.792969 91.792969s-91.792969-41.179688-91.792969-91.792969 41.179688-91.792969 91.792969-91.792969 91.792969 41.179688 91.792969 91.792969zm0 0" />
                </svg>
              </div>
            </div>
            <div className="userCfgPanel" id="userCfgPanel">
              <div className="panelOption config">
                <SecondaryButton
                  value="Ajustes de usuario"
                  action={toggleUserConfig}
                />
              </div>
              <div className="panelOption logOut">
                <SecondaryButton
                  value="Cerrar Sesión"
                  action={disconnect}
                />
                <div id="flag"><strong style={{color: 'crimson', fontSize: '10rem'}}>FLAG</strong></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    }
}

export default Contacts
