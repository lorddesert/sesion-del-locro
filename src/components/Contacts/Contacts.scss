
.Contacts {
  z-index: 1;
  background: #f0f0f0;
  color: #000;
  flex: 2;
  display: flex;
  flex-direction: column;
  .Contacts-content {
    max-width: 20vw;
    overflow: auto;
    flex: 9;
    display: flex;
    flex-direction: column;
    background: #242628;
    // For administrate the white spaces
    // justify-content: space-evenly;

    .notContactsFound {
      display: flex;
      flex-direction: column;
      flex: 1;
      align-items: center;
      justify-content: center;

      span {
        font-size: 1.5em;
        font-family: Verdana, Geneva, Tahoma, sans-serif;
        color: #666;
        opacity: 0.6;
        padding: .5em;
        font-weight: bold;
        font-size: 1.6em;
        line-height: 1.6em;
      }
    }
  }
}

.Profile {
  display: flex;
  justify-content: space-between !important;
  background: #1c1c1c;

  .Profile-img, .Profile-actions {
    padding: .5em;
  }
  .Profile-img {
    display: flex;
    align-items: center;
  }
  .Profile-img img {
    border-radius: 50%;
    width: 50px;
    height: 50px;
  }
  .Profile-actions {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    gap: 5px;
    width: 33%;
  }
  .Action {
      border-radius: 50%;
      display: flex;
      justify-content: center;
      align-items: center;
  }
  .Action-img {
    fill: #DDCF1B;
    width: 20px;
    cursor: pointer;
  }
}
.chat-room {
  display: grid;
  grid-template-columns: 1fr 33%;
  place-items: center start;
  font-size: 1rem;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  gap: 2em;

  color: #DDCF1B;
  background: #1c1c1c;

  span {
    text-align: start;
    padding-left: .5em;
  }

  .image-wrapper {
    display: grid;
    place-items: center;
    cursor: pointer;
    width: 100%;

  }

  img {
    width: 20px;
    height: 20px;
    padding: 1em;

    &:hover {
      background: #48484b;
      border-radius: 50%;
    }
  }

}
#contacts-display {
  display: none;
  flex-direction: column;

}

#userCfg {
  transition: all ease 0.3s;
}

.rotate {
  transition: all ease 0.3s;
  transform: rotate(-90deg);
}
.userCfgPanel {
  display: none;
  place-items: start;
  width: 15vw;
  gap: 5px;
  position: absolute;
  z-index: 8;
  left: 52px;
  top: 66px;
  grid-template-areas: "state"
                       "userConfig"
                       "logOut";
  font-size: 20px;
  background: #49494D;
  opacity: 0;
  transition: 0.1s all ease;
  transform: translateY(-10px);
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  overflow: hidden;
  .SecondaryButton {
    padding: .5em 0 !important;
  }
  button {
    font-size: 1rem;
    color: #fff;
  }
  .panelOption {
    font-size: 1rem;
    margin-right: 1em;
  }

  .userConfigPanel {
    grid-area: userConfig;
  }

  .logOut {
    grid-area: logOut;
    button {
      // color: #da2d56;
      color: #DDCF1B;
      font-weight: bold;
    }
  }
}
.display {
  // transition: 0.1s all ease;
  transform: translateY(0);
  opacity: 1;
}