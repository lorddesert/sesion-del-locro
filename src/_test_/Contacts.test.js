/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom'

import * as React from 'react'
import { render } from '@testing-library/react'
// import renderer from 'react-test-renderer';

import Contacts from'../components/Contacts/Contacts'

  // test('Render component: ', () => {
  //   render(<Contacts />)
  // })

test('Contacts conditional fallback rendering is OK: ', () => {
  const {debug} = render(<Contacts />)
  // className="chat-room"
  // const flag = document.querySelector('#flag')
  
  // expect(flag).toBeTruthy()
  // debug()
})
