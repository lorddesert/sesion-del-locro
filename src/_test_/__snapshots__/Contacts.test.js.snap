// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should'nt have falsy value con contacts state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Contacts"
        id="contacts"
      >
        <div
          class="Contacts-content"
        >
          <div
            class="Profile"
          >
            <div
              class="Profile-img"
            />
            <div
              class="Profile-actions"
            >
              <div
                class="Action"
              >
                <svg
                  class="Action-img"
                  id="userCfg"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m499.953125 197.703125-39.351563-8.554687c-3.421874-10.476563-7.660156-20.695313-12.664062-30.539063l21.785156-33.886719c3.890625-6.054687 3.035156-14.003906-2.050781-19.089844l-61.304687-61.304687c-5.085938-5.085937-13.035157-5.941406-19.089844-2.050781l-33.886719 21.785156c-9.84375-5.003906-20.0625-9.242188-30.539063-12.664062l-8.554687-39.351563c-1.527344-7.03125-7.753906-12.046875-14.949219-12.046875h-86.695312c-7.195313 0-13.421875 5.015625-14.949219 12.046875l-8.554687 39.351563c-10.476563 3.421874-20.695313 7.660156-30.539063 12.664062l-33.886719-21.785156c-6.054687-3.890625-14.003906-3.035156-19.089844 2.050781l-61.304687 61.304687c-5.085937 5.085938-5.941406 13.035157-2.050781 19.089844l21.785156 33.886719c-5.003906 9.84375-9.242188 20.0625-12.664062 30.539063l-39.351563 8.554687c-7.03125 1.53125-12.046875 7.753906-12.046875 14.949219v86.695312c0 7.195313 5.015625 13.417969 12.046875 14.949219l39.351563 8.554687c3.421874 10.476563 7.660156 20.695313 12.664062 30.539063l-21.785156 33.886719c-3.890625 6.054687-3.035156 14.003906 2.050781 19.089844l61.304687 61.304687c5.085938 5.085937 13.035157 5.941406 19.089844 2.050781l33.886719-21.785156c9.84375 5.003906 20.0625 9.242188 30.539063 12.664062l8.554687 39.351563c1.527344 7.03125 7.753906 12.046875 14.949219 12.046875h86.695312c7.195313 0 13.421875-5.015625 14.949219-12.046875l8.554687-39.351563c10.476563-3.421874 20.695313-7.660156 30.539063-12.664062l33.886719 21.785156c6.054687 3.890625 14.003906 3.039063 19.089844-2.050781l61.304687-61.304687c5.085937-5.085938 5.941406-13.035157 2.050781-19.089844l-21.785156-33.886719c5.003906-9.84375 9.242188-20.0625 12.664062-30.539063l39.351563-8.554687c7.03125-1.53125 12.046875-7.753906 12.046875-14.949219v-86.695312c0-7.195313-5.015625-13.417969-12.046875-14.949219zm-152.160156 58.296875c0 50.613281-41.179688 91.792969-91.792969 91.792969s-91.792969-41.179688-91.792969-91.792969 41.179688-91.792969 91.792969-91.792969 91.792969 41.179688 91.792969 91.792969zm0 0"
                  />
                </svg>
              </div>
            </div>
            <div
              class="userCfgPanel"
              id="userCfgPanel"
            >
              <div
                class="panelOption config"
              >
                <div
                  class="SecondaryButton"
                >
                  <button>
                    Ajustes de usuario
                  </button>
                </div>
              </div>
              <div
                class="panelOption logOut"
              >
                <div
                  class="SecondaryButton"
                >
                  <button>
                    Cerrar Sesión
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            id="flag"
            style="padding-left: 5em; transform: rotate(90deg);"
          >
            <strong
              id="flagAnimation"
              style="font-size: 10rem;"
            >
              <span
                id="flag1"
              >
                F
              </span>
              <span
                id="flag2"
              >
                L
              </span>
              <span
                id="flag3"
              >
                A
              </span>
              <span
                id="flag4"
              >
                G
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Contacts"
      id="contacts"
    >
      <div
        class="Contacts-content"
      >
        <div
          class="Profile"
        >
          <div
            class="Profile-img"
          />
          <div
            class="Profile-actions"
          >
            <div
              class="Action"
            >
              <svg
                class="Action-img"
                id="userCfg"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m499.953125 197.703125-39.351563-8.554687c-3.421874-10.476563-7.660156-20.695313-12.664062-30.539063l21.785156-33.886719c3.890625-6.054687 3.035156-14.003906-2.050781-19.089844l-61.304687-61.304687c-5.085938-5.085937-13.035157-5.941406-19.089844-2.050781l-33.886719 21.785156c-9.84375-5.003906-20.0625-9.242188-30.539063-12.664062l-8.554687-39.351563c-1.527344-7.03125-7.753906-12.046875-14.949219-12.046875h-86.695312c-7.195313 0-13.421875 5.015625-14.949219 12.046875l-8.554687 39.351563c-10.476563 3.421874-20.695313 7.660156-30.539063 12.664062l-33.886719-21.785156c-6.054687-3.890625-14.003906-3.035156-19.089844 2.050781l-61.304687 61.304687c-5.085937 5.085938-5.941406 13.035157-2.050781 19.089844l21.785156 33.886719c-5.003906 9.84375-9.242188 20.0625-12.664062 30.539063l-39.351563 8.554687c-7.03125 1.53125-12.046875 7.753906-12.046875 14.949219v86.695312c0 7.195313 5.015625 13.417969 12.046875 14.949219l39.351563 8.554687c3.421874 10.476563 7.660156 20.695313 12.664062 30.539063l-21.785156 33.886719c-3.890625 6.054687-3.035156 14.003906 2.050781 19.089844l61.304687 61.304687c5.085938 5.085937 13.035157 5.941406 19.089844 2.050781l33.886719-21.785156c9.84375 5.003906 20.0625 9.242188 30.539063 12.664062l8.554687 39.351563c1.527344 7.03125 7.753906 12.046875 14.949219 12.046875h86.695312c7.195313 0 13.421875-5.015625 14.949219-12.046875l8.554687-39.351563c10.476563-3.421874 20.695313-7.660156 30.539063-12.664062l33.886719 21.785156c6.054687 3.890625 14.003906 3.039063 19.089844-2.050781l61.304687-61.304687c5.085937-5.085938 5.941406-13.035157 2.050781-19.089844l-21.785156-33.886719c5.003906-9.84375 9.242188-20.0625 12.664062-30.539063l39.351563-8.554687c7.03125-1.53125 12.046875-7.753906 12.046875-14.949219v-86.695312c0-7.195313-5.015625-13.417969-12.046875-14.949219zm-152.160156 58.296875c0 50.613281-41.179688 91.792969-91.792969 91.792969s-91.792969-41.179688-91.792969-91.792969 41.179688-91.792969 91.792969-91.792969 91.792969 41.179688 91.792969 91.792969zm0 0"
                />
              </svg>
            </div>
          </div>
          <div
            class="userCfgPanel"
            id="userCfgPanel"
          >
            <div
              class="panelOption config"
            >
              <div
                class="SecondaryButton"
              >
                <button>
                  Ajustes de usuario
                </button>
              </div>
            </div>
            <div
              class="panelOption logOut"
            >
              <div
                class="SecondaryButton"
              >
                <button>
                  Cerrar Sesión
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          id="flag"
          style="padding-left: 5em; transform: rotate(90deg);"
        >
          <strong
            id="flagAnimation"
            style="font-size: 10rem;"
          >
            <span
              id="flag1"
            >
              F
            </span>
            <span
              id="flag2"
            >
              L
            </span>
            <span
              id="flag3"
            >
              A
            </span>
            <span
              id="flag4"
            >
              G
            </span>
          </strong>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
